### PR TITLE
refactor(router): convert route attributes to plain objects

### PR DESCRIPTION
**Pull request by blackshadev**
# What

@aidan-casey made an alternative suggestion to removing the `Route` as abstract base. With this rework we also lose the `IsRoute` trait and all `Route` builder "verbs" (`Get`, `Post`, etc) become plain old php objects. No method, no logic, no magic. This will make it easier to use the `Route` interface. But it will make it "harder" to do logic in these Route builder "verbs" as you do not have any method to hook into, you should do everything in the constructor. 

Personally, I think this is as clean as it gets.

# DO NOT MERGE

Based on  #821 as PHPStan 2.0 is required, the old one will panic at the sight of property hooks in the interface

### CODE REVIEW

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -9,6 +9,7 @@
     ])
     ->name('*.php')
     ->notName('*.cache.php')
+    ->notPath('Tempest/Router/src/Route.php') // phpcs doesn't yet support property hooks in interfaces
```
Comment (aidan-casey) [inline .php-cs-fixer.dist.php:12]: Simply commenting to say let's keep an eye on [this PR](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8312).

**Comment by aidan-casey** [comment]
Cool PR!

File: src/Tempest/Router/src/Route.php [hunk 2]
```diff
@@ -8,10 +8,9 @@
...
 {
-    public function method(): Method;
+    public Method $method { get; }
 
-    public function uri(): string;
+    public string $uri { get; }
 
-    /** @return class-string<HttpMiddleware>[] */
-    public function middleware(): array;
+    public array $middleware { get; }
```
Comment (brendt) [inline src/Tempest/Router/src/Route.php:15]: Can't you keep the docblock?

**Comment by brendt** [comment]
Nice!

File: src/Tempest/Router/src/Route.php (hunk 2 above)
Comment (blackshadev) [reply src/Tempest/Router/src/Route.php:16]: Yes I forgot, thanks!

**Comment by blackshadev** [comment]
PHPStan seems drunk, can someone rerun it? It reports an error of unnecessary null safe, while on that line (and even in the file) no nullsafe operators are used...

**Comment by aidan-casey** [comment]
@blackshadev I reran it from mobile, but same error. What file/line is it complaining about? It's not complaining locally?

**Comment by blackshadev** [comment]
> @blackshadev I reran it from mobile, but same error. What file/line is it complaining about? It's not complaining locally?

It is not, but I found out why. Somebody other than me rebased this  branch from main , which contained the new error. I rebased the phpstan-v2 branch and added the error to the baseline. I will rebase this one to phpstan-v2 and it should be green.

**Comment by brendt** [comment]
Very nice!

**Comment by brendt** [comment]
Do we need to update the docs as well?

**Comment by blackshadev** [comment]
> Do we need to update the docs as well?

A bit. I can do it tomorrow.

**Comment by blackshadev** [comment]
> Do we need to update the docs as well?

updated [docs in this PR](https://github.com/tempestphp/tempest-docs/pull/48)

**Pull request merged**